pkg/controllers/tlspolicy/tlspolicy_controller.go

```diff
@@ -151,6 +152,24 @@ func (r *TLSPolicyReconciler) reconcileResources(ctx context.Context, tlsPolicy
 		return err
 	}
 
+	// validate that the issuer specified exists
+	issuerName := tlsPolicy.Spec.IssuerRef.Name
+	issuerKind := tlsPolicy.Spec.IssuerRef.Kind
+	if issuerKind == v1.ClusterIssuerKind {
+		issuer := &v1.ClusterIssuer{}
+		err = r.Client().Get(ctx, client.ObjectKey{Name: issuerName}, issuer)
+		if err != nil {
+			return err
+		}
+	}
+	if issuerKind == v1.IssuerKind {
+		issuer := &v1.Issuer{}
+		err = r.Client().Get(ctx, client.ObjectKey{Name: issuerName, Namespace: tlsPolicy.Namespace}, issuer)
+		if err != nil {
+			return err
+		}
+	}
+
```

Comment:
```suggestion
	// validate that the issuer specified exists
	issuerName := tlsPolicy.Spec.IssuerRef.Name
	issuerKind := tlsPolicy.Spec.IssuerRef.Kind
	var issuer client.Object
	issuerNamespace := ""
	if issuerKind == v1.ClusterIssuerKind {
		issuer = &v1.ClusterIssuer{}
	} else if issuerKind == v1.IssuerKind || issuerKind == "" {
		issuer = &v1.Issuer{}
		issuerNamespace = tlsPolicy.Namespace
	} else {
		return fmt.Errorf("issuer kind not supported: %s. Must be either ClusterIssuer or Issuer", issuerKind)
	}

	err = r.Client().Get(ctx, client.ObjectKey{Name: issuerName, Namespace: issuerNamespace}, issuer)
	if err != nil {
		return err
	}

```

I saw in the documentation of `IssuerRef.Kind` that if the value is left empty it's treated as `Issuer`. Also added an extra `else` for the case of any other invalid values, and a tiny bit of refactoring

Reply:
Nice. Pushed. 👍